extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/GoogleTransferExtension.java

```diff
@@ -0,0 +1,69 @@
+package org.dataportabilityproject.datatransfer.google;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.datatransfer.google.calendar.GoogleCalendarExporter;
+import org.dataportabilityproject.datatransfer.google.calendar.GoogleCalendarImporter;
+import org.dataportabilityproject.datatransfer.google.contacts.GoogleContactsExporter;
+import org.dataportabilityproject.datatransfer.google.contacts.GoogleContactsImporter;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+import org.dataportabilityproject.spi.transfer.extension.TransferExtension;
+import org.dataportabilityproject.spi.transfer.provider.Exporter;
+import org.dataportabilityproject.spi.transfer.provider.Importer;
+
+/*
+ * GoogleTransferExtension allows for importers and exporters of data types
+ * to be retrieved.
+ */
+public class GoogleTransferExtension implements TransferExtension {
+  public static final String SERVICE_ID = "google";
+  // TODO: centralized place, or enum type for these
+  private ImmutableList<String> supportedServices = ImmutableList.of("calendar", "contacts");
+  private ImmutableMap<String, Importer> importerMap;
+  private ImmutableMap<String, Exporter> exporterMap;
+  private JobStore jobStore;
+  private boolean initialized = false;
+
+  @Override
+  public String getServiceId() {
+    return SERVICE_ID;
+  }
+
+  @Override
+  public Exporter<?, ?> getExporter(String transferDataType) {
+    Preconditions.checkArgument(initialized);
+    Preconditions.checkArgument(supportedServices.contains(transferDataType));
+    return exporterMap.get(transferDataType);
+  }
+
+  @Override
+  public Importer<?, ?> getImporter(String transferDataType) {
+    Preconditions.checkArgument(initialized);
+    Preconditions.checkArgument(supportedServices.contains(transferDataType));
+    return importerMap.get(transferDataType);
+  }
+
+  @Override
+  public void initialize(ExtensionContext context) {
+    // Note: initialize could be called twice in an account migration scenario where we import and
+    // export to the same service provider. So just return rather than throwing if called multiple
+    // times.
+    if(initialized) return;
+
+    jobStore = context.getService(JobStore.class);
+
+    ImmutableMap.Builder<String, Importer> importerBuilder = ImmutableMap.builder();
+    importerBuilder.put("contacts", new GoogleContactsImporter());
```

Comment:
This is not strictly necessary but would be easy-ish and nice to do. Can be a TODO if you want:

We're using Guice for the Google CloudExtension and would be cool to do for our transfer extension as well. You'd need to annotate GoogleContactsImporter and Exporter's constructors with @Inject, and include a GoogleTransferExtensionModule in this directory, which would @Provide a PeopleService. (Actually I think the no-arg constructors you're calling will initialize the peopleService and calendarInterface's to null and not work, maybe we should get rid of those actually)

You can look in GoogleCloudExtension as an example, I think it'd be pretty easy but it's fine with me to leave as is too.

Reply:
Going to approve and merge for now and we can discuss and change if we want in a future PR